frontend/src/v5/ui/components/viewer/drawingViewer/viewerLayer2D/camera/cameraOffSight.component.tsx

```diff
@@ -0,0 +1,97 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */ 
+import { DrawingsHooksSelectors } from '@/v5/services/selectorsHooks';
+import { useSearchParam } from '@/v5/ui/routes/useSearchParam';
+
+import { clamp } from 'lodash';
+import { useRef, useState, useEffect } from 'react';
+import { useParams } from 'react-router';
+import { Vector2 } from 'three';
+import { useModelLoading, useViewpointSubscription } from './viewer.hooks';
+import { CameraOffSightIcon } from './camera.styles';
+import { ViewerParams } from '@/v5/ui/routes/routes.constants';
+
+interface Props {
+	onCameraSightChanged: (cameraOnSight: boolean) => void;
+	scale: any;
+	viewport: any;
+}
+
+
+const cameraPadding = 3;
+const iconSize = 56;
+
+export const CameraOffSight = ({ onCameraSightChanged, viewport, scale }: Props) => {
+	const viewpoint = useRef(null);
+	const viewportRef = useRef(viewport);
+	const camInSight = useRef(false);
+	const scaleRef = useRef(scale);
+
+	const [drawingId] = useSearchParam('drawingId');
+	const { containerOrFederation } = useParams<ViewerParams>();
+	const [position, setPosition] = useState({ x:0, y:0 });
+	const [angle, setAngle] = useState(0);
+	const transform2DTo3D = DrawingsHooksSelectors.selectTransform2Dto3D(drawingId, containerOrFederation);
+	const transform3DTo2D = DrawingsHooksSelectors.selectTransform3Dto2D(drawingId, containerOrFederation);
+	const modelLoading = useModelLoading();
+
+	useEffect(() => viewportRef.current = viewport, [viewport]);
+	useEffect(() => scaleRef.current = scale, [scale]);
+
+	useViewpointSubscription((v) => {
```

Comment:
The camera position can not change but pan and zoom does so this have to be updated regardless. And because this doesnt seem to impact the performance and the code is more simple this way I dont see much merit to change it.